app/containers/HomePage/slice.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * HomePage Slice
+ *
+ * Here we define:
+ * - The shape of our container's slice of Redux store,
+ * - All the actions which can be triggered for this slice, including their effects on the store.
+ *
+ * Note that, while we are using dot notation in our reducer, we are not actually mutating the state
+ * manually. Under the hood, we use immer to apply these updates to a new copy of the state.
+ * Please see https://immerjs.github.io/immer/docs/introduction for more information.
+ *
+ */
+
+import { createSlice } from '@reduxjs/toolkit';
+
+// The initial state of the HomePage container
+export const initialState = {
+  username: '',
+  repositories: [],
+  loading: false,
+  error: false,
+};
+
+const slice = createSlice({
```

Comment:
Maybe this is a bigger conversation that should be resolved outside this PR, but I've found doing redux per-page like this makes it hard to re-use state on another page without an awkward coupling between pages. 

i.e. If I need to access the repos on a different page, say "ReposPage", then I have to either duplicate this slice, or import this slice. Then you end up with ReposPage importing from HomePage, which feels awkward.

Alternatively, I believe it's common to have a "features" folder. This could have a "repos" slice; then any number of pages can import from this slice. Another option is what I described in [this comment](https://github.com/react-boilerplate/react-boilerplate/issues/2806#issuecomment-555195354), but that's kind of opinionated and not common.

Reply:
Yeah, notionally the state tree and the component tree are meant to be somewhat independent of each other.  Any number of components could be depending on a given chunk of data in the store, and the state tree isn't necessarily going to be shaped the same way as the component tree.

I've seen some folks talk about having "data layer" folders which just have Redux logic in them, and then there _may_ be more Redux logic with specific "pages" or "containers" if appropriate.  That sounds similar to what you're describing, I think.

That said, I personally don't have any qualms about different slices importing each other in terms of "privacy" or anything.  You're basically acknowledging that there is an explicit dependency between these different parts of the application logic, and that not everything is going to be perfectly isolated and encapsulated.  Note that if you take that far enough, [it _is_ possible to end up with circular reference import problems](https://redux-toolkit.js.org/usage/usage-guide/#exporting-and-using-slices).

Organizing information is _hard_ :)

Reply:
Thanks for the comments guys! Makes a lot of sense. Here's my summary:

**What we had** Repos management happening inside the App container. Injected as `globalState` and `globalReducer`. This felt clunky to me because it could grow infinitely and I don't think we explain this pattern anywhere in the docs anyway.

**What I tried to do:** Move repos management to the HomePage. Admittedly not a great idea. For all the reasons mentioned above.

**What I ended up doing in [my last commit](https://github.com/react-boilerplate/react-boilerplate/pull/2808/commits/ce63b7e0134ae01504474ac19cd6580813a812f5):** Move repos management to a ReposManager as described by @BenLorantfy. Didn't go all the way to create a `managers` folder because that should indeed be a bigger conversation with accompanying doc updates. I just wanted to at least move repos management outside of the App container. ReposManager is still rendered inside the App (i.e. "globally") so functionality hasn't changed.

Makes sense to everyone? If yes, please re-approve? 🙏 I've updated the base branch to `redux-toolkit-switch` so that we can merge this project in full once it's done.

CC @jwinn @justingreenberg

Reply:
Oops no need for re-approvals. Will wait to merge in case anyone has comments though.

Reply:
@julienben this is awesome, much cleaner 👏 👍 LGTM!!

Reply:
@julienben great work!

Reply:
Looks great 👍